### PR TITLE
Fix union type suggestions

### PR DESCRIPTION
**Pull request by kanushka**
## Summary
- Fix type suggestion filtering for union type members in BI forms.
- Filter visible type completions by the active union member instead of the whole type expression.
- Share the filtering helper from the side-panel package for both flow node and artifact forms.

Fixes wso2/product-integrator#1130

https://github.com/user-attachments/assets/5af47246-bce3-418a-9e99-ddf07d974407

## Root cause
After typing a union delimiter such as int pipe, the client filtered cached visible types using the full text before the cursor. That produced a query containing the delimiter, so no type label matched and suggestions disappeared.

## Validation
- rush build --to @wso2/ballerina-side-panel
- rush build --to @wso2/ballerina-visualizer

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced type completion filtering to properly handle union type members and context-aware text extraction, improving accuracy when filtering types in union declarations.

* **Tests**
  * Introduced comprehensive test coverage for type completion utilities, validating edge cases including union member selection, whitespace handling, and cursor positioning scenarios.

### CODE REVIEW

**Comment by NipunaRanasinghe** [comment]
@coderabbitai review

File: workspaces/ballerina/ballerina-side-panel/package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "types": "lib/index.d.ts",
     "scripts": {
         "build": "tsc --pretty && npm run copy:assets",
+        "test": "npm run build && node --experimental-default-type=module --test test/typeCompletionUtils.test.js",
```
[the remark this answers is not in the document]
Comment (kanushka) [reply workspaces/ballerina/ballerina-side-panel/package.json]: Fixed with https://github.com/wso2/vscode-extensions/pull/2097/commits/f93b7a58bd30e1628b31823a487aac6f6e44f0ff

File: workspaces/ballerina/ballerina-side-panel/test/typeCompletionUtils.test.mjs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import test from "node:test";
+import assert from "node:assert/strict";
+
+import { getTypeCompletionSearchText } from "../lib/components/editors/typeCompletionUtils.js";
```
[the remark this answers is not in the document]
Comment (kanushka) [reply workspaces/ballerina/ballerina-side-panel/test/typeCompletionUtils.test.mjs:4]: We have not exported util from index yet. will do this in another refactoring

**Review by axewilledge — APPROVED** [review]
